app/Middleware/Queue/SubmissionService.php

```diff
@@ -167,10 +167,36 @@ public function getConsumerName()
     public function doSubmit(Message $message)
     {
         try {
-            $fh = fopen($message->file, 'r');
             $config = Config::getInstance();
             $config->set('CDASH_REMOTE_ADDR', $message->ip);
-            do_submit($fh, $message->project, null, $message->md5, $message->checksum);
+
+            $local_execution = file_exists($message->file);
+            if ($local_execution) {
+                // Local execution, process an open file.
+                $fh = fopen($message->file, 'r');
+            } else {
+                // Remote execution, pass the filename and CDash will retrieve
+                // its contents.
+                $fh = basename($message->file);
+            }
+            $handler = do_submit($fh, $message->project, null, $message->md5, $message->checksum);
+
+            if (!$local_execution) {
+                // If remotely executing, tell the server to delete the file
+                // now that it has been parsed.
+                $query_args = ['filename' => basename($message->file)];
+                if (is_object($handler) && property_exists($handler, 'backupFileName')) {
+                    // If the handler provided us with a descriptive backup
+                    // filename we pass this on to the server in case it is
+                    // configured to rename (rather than delete) processed
+                    // submission files.
+                    $query_args['dest'] = basename($handler->backupFileName);
+                }
+                $client = new \GuzzleHttp\Client();
```

Comment:
We should get into the habit of always injecting our dependencies. That is unless the method's specific purpose is that of object creation (in which case it should still check first to ensure that an instantiation of the property does not already exist). You can either use lazy load accessors as in:
```php
function getHttpClient()
{
  if (!$this->httpClient) {
    $this->httpClient = new \GuzzleHttp\Client();
  }
  return $this->httpClient;
}

function setHttpClient(HttpClientInterface $client)
{
  $this->httpClient = $client
}
```
or use PHPDI, as in:
```php
$container = ServiceContainer::getInstance();
$httpClient = $container->make(HttpClientInterface::class);
```
or you could do DI the old fashioned way and create a constructor, `SubmissionService::__construct(HttpClientInterface $client)`